_field-types/supported-field-types/knn-vector.md

```diff
@@ -163,4 +166,102 @@ GET test-index/_search
   }
 }
 ```
-{% include copy-curl.html %}
\ No newline at end of file
+{% include copy-curl.html %}
+
+### Quantization techniques
+
+If your vectors are of type `float`, you need to first convert them to `byte` before ingesting the documents. This conversion is accomplished by _quantizing the dataset_---reducing the precision of its vectors. There are many quantization techniques, such as scalar quantization or product quantization (PQ), which is used in the Faiss engine. The choice of quantization technique depends on the type of data you're using and can affect the accuracy of recall values. The following sections describe the scalar quantization algorithms that were used to quantize the [k-NN benchmarking test](https://github.com/opensearch-project/k-NN/tree/main/benchmarks/perf-tool) data for the [L2](#scalar-quantization-for-the-l2-space-type) and [cosine similarity](#scalar-quantization-for-the-cosine-similarity-space-type) space types.
+
+#### Scalar quantization for the L2 space type
+
+For Euclidean datasets, we recommend using a scalar quantization technique with L2 space type because Euclidean distance is shift invariant. If you shift both $$x$$ and $$y$$ by the same $$z$$ then the distance remains the same, which means $$\lVert x-y\rVert =\lVert (x-z)-(y-z)\rVert$$.
+
+The following example pseudocode illustrates scalar quantization for the L2 space type:
```

Comment:
We shouldn't explain it this way. Can you rephrase it to something like this:

The following example pseudocode illustrates the scalar quantization technique used for the benchmarking tests on Euclidean Datasets (uses L2 space type). Euclidean distance is shift invariant, if you shift both $$x$$ and $$y$$ by the same $$z$$ then the distance remains the same, which means $$\lVert x-y\rVert =\lVert (x-z)-(y-z)\rVert$$.